Верстка

gin-az opened the pull request.

gin-az committed:
Ok

Okay

gin-az committed:
Update style.css

gin-az committed:
edit index2

Скорректирован index2.html

gin-az committed:
1

gin-az reviewed (commented):
Добавлен пробел

Roman-Maksimov commented on index.html:
```diff
@@ -5,8 +5,202 @@
     <title>Document</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
```
В данном случае принципиальной разницы нет, но рекомендую всегда помещать вашу таблицу стилей последней. Так у вас будет возможность переопределять third-party стили.

Roman-Maksimov commented on index.html:
```diff
@@ -5,8 +5,202 @@
+        </div>
+        <div class="block-header-icons">
+            <object class="item-header-icons" type="image/svg+xml" data="shopping_cart.svg">
+                <img src="img\shopping_cart.svg">
```
img - самозакрывающийся тэг

Roman-Maksimov commented on index2.html:
```diff
@@ -0,0 +1,51 @@
+    <title>Document</title>
+    <link href="style.css" rel="stylesheet" type="text/css">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
```
В данном случае принципиальной разницы нет, но рекомендую всегда помещать вашу таблицу стилей последней. Так у вас будет возможность переопределять third-party стили.

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+    background: transparent; 
+    background-image: url("./img/favorite_border.svg");
+     background-repeat: no-repeat;
+    border: none !important;
```
!important -  плохо, надо придумать что-нибудь другое, например, усилить специфичность.

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+.minus, .plus {
+    box-sizing: border-box;
+	width: 40px;
+    border: none !important;
```
!important

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+    background: transparent; 
+    background-image: url("./img/favorite_border.svg");
+    background-repeat: no-repeat;
+    border: none !important;
```
!important

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+    height: 26px;
+    width: 38px;
+    background: transparent; 
+    border: none !important;
```
!important

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+.block_product {
+    display: flex;
+    width: 100%;
+     flex-wrap: wrap;  
```
`flex-wrap` тут не нужен. Лучше `flex-direction: column`.

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,336 @@
+} 
+
+@media only screen 
+and (min-device-width : 365px) 
```
min-width/max-width. min-device-width работает не на всех устройствах.

Roman-Maksimov replied:
А что с окнами уже 365px?

Roman-Maksimov reviewed (changes requested):
Следите за code-style. Строчки прыгают.

gin-az committed:
Внесены исправления по замечаниям.

1) Code-style.
2) Изменен порядок подключения шрифта и таблицы стилей.
3) Удалены !import.
4) Исправлены стили для iphone 11.

Roman-Maksimov commented on index.html:
```diff
@@ -5,8 +5,201 @@
+        </div>
+        <div class="block-header-icons">
+            <object class="item-header-icons" type="image/svg+xml" data="shopping_cart.svg">
+            <img src="img\shopping_cart.svg">
```
img тэги надо закрыть

Roman-Maksimov commented on script.js:
```diff
@@ -0,0 +1,30 @@
+
+    likeBtn.addEventListener("click", function()
+        {
+          if (likeBtn.classList == "like-button")
```
В подавляющем большинстве случаев рекомендуется использовать тождественное сравнение `===`

Roman-Maksimov replied:
Что если у кнопки будут еще какие-то классы кроме этого? Ниже вы используете методы `ClassList`,  здесь тоже имеет смысл использовать `classList.contains`

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+    padding-left: 0;
+} 
+
+@media only screen and (min-width : 365px) and (max-width : 667px){ 
```
Всё же, как быть с устройствами уже 365px, например iPhone 5 - 320px? Вы хотели показать одну карточку на очень узких окнах?

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+@media only screen and (min-width : 365px) and (max-width : 667px){ 
+
+.item {
+    flex: 0 0 155px; 
```
При ширине окна 530px на странице видно 3 карточки. Во-первых, в мобильном варианте ширину карточки не обязательно фиксировать, она может уменьшаться (flex-shrink). Во-вторых, 667px на странице каталога явно многовато для перехода на мобильную верстку (2х220 = 440px + паддинги), а для детальной страницы - маловато: картинка(300px) + block_price(~458px) + margin(2х24px) = 806px.

PS. Не обязательно делать один media, можно сделать несколько, например, по одному на каждую страницу.

Roman-Maksimov commented on index2.html:
```diff
@@ -0,0 +1,49 @@
+                <input type="number" min="1" max="100" step="1" value="1"class="num"/></td>
+                <button class="plus"> + </button>
+            </div>    
+            <div><img src="img\add_to_cart.svg"> </div>
```
Не лучший вариант. Здесь должен быть `button`. Стили там не сложные. Так же у кнопки есть состояние `:hover` (там добавляется тень и немного меняется цвет) + анимация

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+    -moz-box-shadow: inset 0px -70px 45px 0px rgba(0, 0, 0, 0.15);
+    -o-box-shadow: inset 0px -70px 45px 0px rgba(0, 0, 0, 0.15);
+    box-shadow: inset 0px -70px 45px 0px rgba(0, 0, 0, 0.15);
+    box-shadow: inset 0px -70px 45px 0px rgba(0, 0, 0, 0.15);
```
1. Дублированная строка
2. По макету здесь градиент, а не тень (в макете `IPhone 11 Pro / X - detalied page` -> `Image` -> `Rectangle 8` -> на вкладке `Inspect` он уже готовый, достаточно просто скопировать)

Roman-Maksimov replied:
Блок с картинкой тут должен быть растянут на всю ширину окна.

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+
+.item {
+    flex: 0 0 155px; 
+    width: 45%;
```
У вас `flex-basis: 155px`. `width` тут ни на что не влияет.

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+
+.catalog{
+    justify-content: center;
+    padding: 2px 1px 0px;
```
откуда такие странные паддинги?

Roman-Maksimov commented on style.css:
```diff
@@ -0,0 +1,335 @@
+}
+.block_price{
+    flex-wrap: wrap;
+}
```
Обратите внимание, по макету блок с ценой и кнопками в мобильном варианте должен быть в две строки, в одну не влезает. Сверху цена и favorite, снизу кнопки.

gin-az committed:
Исправлены замечания

Замечания исправлены в части стилей и верстки.

Roman-Maksimov commented on media.css:
```diff
@@ -0,0 +1,49 @@
+@media only screen and (min-width: 300px) and (max-width: 490px) {
```
В вашем вариант `max-width 520px` подойдет лучше. При окне меньше 519px две карточки уже не влезают в строку.

Roman-Maksimov commented on media.css:
```diff
@@ -0,0 +1,49 @@
+      flex-shrink: 1;
+      height: 272px;
+      position: relative;
+    }
```
В основном стиле у вас `flex-basis: 220px;`. Это не позволяет сделать ширину карточки адаптивной. В вашем случае нужно что-то вроде:
```css
flex-shrink: 1;
flex-basis: calc(50%-16px);
```
или
```css
flex: 0 1 calc(50% - 16px);
```

gin-az committed:
Внесение исправлений

Внесены правки в стиле для мобильного вида.

Roman-Maksimov commented on media.css:
```diff
@@ -0,0 +1,49 @@
+@media only screen and (min-width: 300px) and (max-width: 520px) {
+    .item {
+      flex-shrink: 1;
```
`flex-shrink` больше не нужен, раз вы указываете общее свойство `flex`
```css
flex: <flex-grow> <flex-shrink> <flex-basis>;
```

gin-az committed:
Поправка в media.css
